mne/channels/interpolation.py

```diff
@@ -251,25 +273,94 @@ def _interpolate_bads_nirs(inst, method="nearest", exclude=(), verbose=None):
     chs = [inst.info["chs"][i] for i in picks_nirs]
     locs3d = np.array([ch["loc"][:3] for ch in chs])
 
-    _check_option("fnirs_method", method, ["nearest", "nan"])
-
-    if method == "nearest":
-        dist = pdist(locs3d)
-        dist = squareform(dist)
-
-        for bad in picks_bad:
-            dists_to_bad = dist[bad]
-            # Ignore distances to self
-            dists_to_bad[dists_to_bad == 0] = np.inf
-            # Ignore distances to other bad channels
-            dists_to_bad[bads_mask] = np.inf
-            # Find closest remaining channels for same frequency
-            closest_idx = np.argmin(dists_to_bad) + (bad % 2)
-            inst._data[bad] = inst._data[closest_idx]
-    else:
-        assert method == "nan"
-        inst._data[picks_bad] = np.nan
+    dist = pdist(locs3d)
+    dist = squareform(dist)
+
+    for bad in picks_bad:
+        dists_to_bad = dist[bad]
+        # Ignore distances to self
+        dists_to_bad[dists_to_bad == 0] = np.inf
+        # Ignore distances to other bad channels
+        dists_to_bad[bads_mask] = np.inf
+        # Find closest remaining channels for same frequency
+        closest_idx = np.argmin(dists_to_bad) + (bad % 2)
+        inst._data[bad] = inst._data[closest_idx]
+
     # TODO: this seems like a bug because it does not respect reset_bads
     inst.info["bads"] = [ch for ch in inst.info["bads"] if ch in exclude]
 
     return inst
+
+
+@verbose
+def _interpolate_bads_seeg(inst, exclude=None, tol=2e-3, verbose=None):
+    if exclude is None:
+        exclude = list()
+    bads_idx = np.zeros(len(inst.ch_names), dtype=bool)
+    goods_idx = np.zeros(len(inst.ch_names), dtype=bool)
+
+    picks = pick_types(inst.info, meg=False, seeg=True, exclude=exclude)
+    inst.info._check_consistency()
+    bads_idx[picks] = [inst.ch_names[ch] in inst.info["bads"] for ch in picks]
+
+    if len(picks) < 3 or bads_idx.sum() == 0:
```

Comment:
[question] IDK how much of an edge case this is or if it's a realistic worry, but should we `warn` if `len(picks) < 3`?  I can imagine maybe being surprised that the single bad channel wasn't interpolated because I (inadvertantly?) excluded all but one of the good channels (leaving just 2 channels in `picks`)

Reply:
Yeah, I think maybe best just to kick it to the runtime error if there's a bad channel on a shaft with less than 3 contacts which is the edge case and I think an error is warranted there since that's not a good interpolation situation. So no need to silently return.

Reply:
I think a warning is better. If you have 3 bad channels and this edge-case represents only one of them, the interpolation will fail and raise unless you mark the edge-case bad channel as good, interpolate, and mark it again as bad. I find this more error prone than warning, interpolating the 2 other channels and ignoring the edge-case one, and finally removing only the 2 interpolated channels from `.info["bads"]`.

Reply:
This won't raise an error if there are 3 bad channels on a shaft (electrode) only if there aren't three good ones. If there are only two good channels, I don't think there is a use case where you want to use those to interpolate any of the bad channels so I'd still be disposed to an error. Although if you had 3 contacts (usually 4 for DBS) but as in a DBS and you wanted to interpolate the middle one on some epochs that would make sense so maybe it should just be two required.

Reply:
I mixed up bad channels/good ones in my comment. I agree there is no use case for interpolating along a shaft if you lack the information to do so. But I would still warn, else if you have 2 shafts each with some bad channels, one with less than 3 good contacts remaining; the interpolation function will fail even for the shaft which has enough information to interpolate.

```
raw.ch_names = [f"Shaft1_{k} for k in range(5)] + [f"Shaft2_{k} for k in range(5)]
raw.info["bads"] = [f"Shaft1_{k} for k in range(4)] + ["Shaft2_0"]
raw.interpolate_bads(...)  # warning for the bad contacts on Shaft1, interpolation for Shaft2_0
assert raw.info["bads"] == [f"Shaft1_{k} for k in range(4)]
```

Or am I misunderstanding something and/or a multi-shaft recording is not realistic?

Reply:
The length of the shaft isn't checked if the channels aren't marked bad so I think if you do mark it and try and interpolate it you should get an error rather than unreliable data. You would then have to drop that channel without interpolation to continue without an error which I think is the safest way to go and wouldn't cause major issues in multi-subject pipelines.

Reply:
I think the other consideration if it's a warning is what do you do to the data in that channel? If it was supposed to be interpolated and you just warned and the data was left as is, that seems potentially misleading. If it only requires two other contacts, I think it's pretty safe that this is an unlikely edge case and the failure will be warranted so that the user can be notified to fix their data by dropping all the contacts on that electrode shaft.